### PR TITLE
example for the new non linear terms

### PR DESCRIPTION
**Pull request by peppe988**
Hi there. I am pulling a simple example for the new non-linear terms I created a few months ago. It was done by modifying the example here: https://sfepy.org/doc-devel/examples/diffusion-poisson_field_dependent_material.html so let me know if this is ok or if you prefer another example or a better description :)

### CODE REVIEW

**Comment by rc** [comment]
Thanks, it works! Can it be tweaked so that the non-linear effect is more visible?

What about naming the file simply `poisson_nonlinear_material.py`?

File: sfepy/examples/diffusion/Non_linear_general_poisson_equation.py [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+    'ic' : ('Omega', {'T.0' : 0.0}),
+}
+
+integrals = {
+    'i' : 1,
+}
+
+equations = {
+    'Temperature' : """dw_nl_diffusion.i.Omega(get_conductivity, d_get_conductivity, s, T ) + dw_volume_nvf.i.Omega(nl_src, d_nl_src,s,T) = 0"""
+}
```
Comment (rc) [inline sfepy/examples/diffusion/Non_linear_general_poisson_equation.py:89]:
```
equations = {
    'Temperature' : """
       dw_nl_diffusion.i.Omega(get_conductivity, d_get_conductivity, s, T)
     + dw_volume_nvf.i.Omega(nl_src, d_nl_src, s, T) = 0
     """
}
``

File: sfepy/examples/diffusion/Non_linear_general_poisson_equation.py [hunk 2]
```diff
@@ -0,0 +1,104 @@
+r"""
+Example usage of the non linear diffusion and non linear volume terms.
+
+The example is an adaptation of:
+https://sfepy.org/doc-devel/examples/diffusion-poisson_field_dependent_material.html
```
Comment (rc) [inline sfepy/examples/diffusion/Non_linear_general_poisson_equation.py:5]: Use
```
:ref:`diffusion-poisson_field_dependent_material`
```

File: sfepy/examples/diffusion/Non_linear_general_poisson_equation.py [hunk 3]
```diff
@@ -0,0 +1,104 @@
+r"""
+Example usage of the non linear diffusion and non linear volume terms.
```
Comment (rc) [inline sfepy/examples/diffusion/Non_linear_general_poisson_equation.py:2]: Nonlinear Poisson's equation example demonstrating the nonlinear diffusion and nonlinear volume force terms.

File: sfepy/examples/diffusion/Non_linear_general_poisson_equation.py [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+https://sfepy.org/doc-devel/examples/diffusion-poisson_field_dependent_material.html
+
+Find :math:`T(t)` for :math:`t \in [0, t_{\rm final}]` such that:
+
+.. math::
+   \int_{\Omega} c(T) \nabla s \cdot \nabla T + \int_{\Omega} g(T) \cdot s
+    = 0
+    \;, \quad \forall s \;.
+
+where :math:`c(T) and g(T)` are the :math:`T` dependent coefficients.
```
Comment (rc) [inline sfepy/examples/diffusion/Non_linear_general_poisson_equation.py:14]:
```
:math:`c(T)` and :math:`g(T)` 
```

**Comment by peppe988** [comment]
now the non-linearity should be more visible

**Comment by rc** [comment]
Thanks, looks good! I am thinking about modifying the nonlinearity also in `poisson_field_dependent_material.py` example? I wonder whether that would even converge...

**Comment by peppe988** [comment]
I completed the last suggestion. 

> Thanks, looks good! I am thinking about modifying the nonlinearity also in `poisson_field_dependent_material.py` example? I wonder whether that would even converge...

Yes, it is very possible, do you want me to test it?

File: sfepy/examples/diffusion/poisson_nonlinear_material.py [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+
+filename_mesh = data_dir + '/meshes/3d/cylinder.mesh'
+
+def get_conductivity(T):
+    """
+    Calculates the conductivity as (1 + T**4) and returns it.
+    """
+    val = 1 + T**4
+    return val
+
```
Comment (rc) [inline sfepy/examples/diffusion/poisson_nonlinear_material.py:29]: I can see an extra whitespace on this line. Maybe set your editor to mark or remove the trailing whitespace.

File: sfepy/examples/diffusion/poisson_nonlinear_material.py [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+     + dw_volume_nvf.i.Omega(nl_src, d_nl_src, s, T) = 0
+     """
+}
+
+solvers = {
+    'ls' : ('ls.scipy_direct', {}),
+    'newton' : ('nls.newton', {
+        'i_max' : 15,
+        'eps_a' : 1e-10,
+        'eps_r' : 1.0,
```
Comment (rc) [inline sfepy/examples/diffusion/poisson_nonlinear_material.py:100]: Tip: Adding
```
        'ls_on' : 10,
```
here results in one less iteration.

**Comment by rc** [comment]
> I completed the last suggestion.
> 
> > Thanks, looks good! I am thinking about modifying the nonlinearity also in `poisson_field_dependent_material.py` example? I wonder whether that would even converge...
> 
> Yes, it is very possible, do you want me to test it?

If you have spare time... There is no source term in that example, so they are not directly comparable anyway.

**Comment by rc** [comment]
Thanks, LGTM!

**Comment by peppe988** [comment]
perfect :)

**Pull request merged**